### DISCUSSION TITLE
How to compile and run yourself?

### DISCUSSION THREAD

**Discussion by henxdl** [legendary-gl/legendary#669] [Q&A] [answered]
I have made a few changes to cli.py that aren't suitable for everyone but that I want to use. How do I compile legendary into an exe to run it? Thanks!

(I haven't ever used github discussions before so I don't know any normal conventions or other things like that, and I'm sorry if this question has already been answered somewhere else ;-;)

**Comment by CommandMC** [comment, marked as answer]
You can either run the `cli.py` file directly by running `PYTHONPATH=. python legendary/cli.py any other args here`, or build an executable with `pyinstaller --onefile --name legendary -i assets/windows_icon.ico legendary/cli.py`

**Reply by henxdl** [reply]
Oh! I didn't know that cli.py was the only requirement needed to compile. Thanks!
